### PR TITLE
Feature/62 get daterange

### LINKED ISSUE #62: Get bookmarks for a date range
Provided with a start and end date, return back the bookmarks that fit in this range

### PR DESCRIPTION
**Pull request by ctocco**
get bookmarks between a range of dates example query 
/bookmarks/date/?startDate=2018.12.01&endDate=2019.05.05
/bookmarks/date/?startDate=2018.12.01
closes #62 #12

### CODE REVIEW

**Comment by kruemmelkhanh** [comment]
i think we need to change the route for this one

File: controller/bookmark.js [hunk 1]
```diff
@@ -81,6 +82,45 @@ module.exports = {
     next()
   },
 
+  // @route   GET api/bookmarks/tag/// /dates/?startDate=2018.12.01&endDate=2019.05.15,
```
Comment (xqwtsz) [inline controller/bookmark.js]: this route desc looks a bit confused 😛

File: helpers/dateParser.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// This function parses the date format from the user
+// user input query: /bookmarks?startDate=2018.12.01&endDate=2019.05.15
```
Comment (xqwtsz) [inline helpers/dateParser.js]: this comment also doesn't match the current route

Comment (ctocco) [reply helpers/dateParser.js]: Thank for that I have fixed the commenting!

File: controller/bookmark.js (hunk 1 above)
Comment (ctocco) [reply controller/bookmark.js]: Thank for that I have fixed the commenting!

**Comment by ctocco** [comment]
Thanks, will get to work on the testing of this function.

**Pull request merged**